### DISCUSSION TITLE
-t epub3 from markdown produces "invalid ids" as reported by calibre

### DISCUSSION THREAD

**Discussion by georgernstgraf** [jgm/pandoc#9412] [Q&A] [answered]
Greetings folks,

These invalid ids are created from many headings i have in my documents. Calibre
offers a "fix automatically" button which then replaces all those ids with
randomly generated ones, and then the check yields no errors anymore.

As asked, I am not filing an issue yet, but I think this here is worth one.

From my Makefile:

```Makefile
ebook.epub::
	pandoc --metadata-file=metadata.yaml --toc --toc-depth=1 -t epub3 -i [0-9][0-9][0-9]*md -o $@
```

See my screenshot:

![Screenshot from 2024-02-05 11-40-36](https://github.com/jgm/pandoc/assets/115418377/cbad5fc1-3749-4a63-9e5d-84800f42f43e)

**Comment by jgm** [comment, marked as answer]
As I understand it, these are not invalid IDs.  epubcheck will not flag them as such.
The epub spec calls for XML-serialized HTML5, and characters like `ü` are allowed in HTML5 identifiers.  It may be that there are readers that have trouble with them, but that is different from claiming that they are "invalid."

If this is a serious problem, though, we could consider adjusting our ids to be ASCII only.  Comments welcome.

**Reply by georgernstgraf** [reply]
Thank you! Importing the generated ebook into my tolino shine <https://mytolino.com/products/tolino-shine-4/> (I have a slightly older model but with very recent firmware) also generates a warning, but I am able to open and read the content.

So no, not serious.

**Reply by jgm** [reply]
Are you generating the epub from pandoc's markdown?  If so, try with the `ascii_identifiers` extension: `pandoc -f markdown+ascii_identifiers -t epub3`.
I think this will solve the problem.

**Reply by georgernstgraf** [reply]
Yes it does .. sorry I missed that. Awesome work!
